worker/caasoperator/caasoperator.go

```diff
@@ -189,6 +189,18 @@ func (op *caasOperator) makeAgentSymlinks(unitTag names.UnitTag) error {
 		return errors.Trace(err)
 	}
 
+	// TODO(caas) - remove this when upstream charmhelpers are fixed
+	// Charmhelpers expect to see a jujud in a machine-X directory.
+	legacyMachineDir := filepath.Join(agentBinaryDir, "machine-0")
+	err = os.Mkdir(legacyMachineDir, 0600)
+	if err != nil && !os.IsExist(err) {
+		return errors.Trace(err)
+	}
+	err = symlink.New(jujudPath, filepath.Join(legacyMachineDir, jujunames.Jujud))
+	if err != nil && !os.IsExist(err) && !os.IsPermission(err) {
```

Comment:
Why the permission clause here? Wouldn't this code be running as root? (Oh, I guess it wouldn't be in tests, is that why?)

Reply:
Yeah, it's a tests thing.

Reply:
I had to do the same thing for audit logging, now that I think about it.